compiler/rustc_mir_build/src/check_unsafety.rs

```diff
@@ -100,6 +127,7 @@ impl<'tcx> UnsafetyVisitor<'_, 'tcx> {
                 *used = true;
             }
             SafetyContext::UnsafeFn if unsafe_op_in_unsafe_fn_allowed => {}
+            SafetyContext::UnsafeFn if self.emit_deprecated_safe_fn_call(span, &kind) => {}
```

Comment:
Code-style-wise, a match guard like this suggests that the function has no side-effects. I initially read the like as "if we should emit a lint related to a call to a deprecated safe function". I'd find it clearer as e.g.:
```suggestion
            SafetyContext::UnsafeFn => {
                let deprecated_safe_fn = self.emit_deprecated_safe_fn_call(span, &kind);
                if !deprecated_safe_fn {
                    // unsafe_op_in_unsafe_fn is disallowed
                	kind.emit_unsafe_op_in_unsafe_fn_lint(
                	    ...
                	)
                }
            }
```

Reply:
Applied your suggestion.